modules/autofill/autofill-api/src/main/java/com/duckduckgo/autofill/store/AutofillStore.kt

```diff
@@ -24,9 +24,19 @@ interface AutofillStore {
     suspend fun getCredentials(rawUrl: String): List<LoginCredentials>
 
     suspend fun saveCredentials(rawUrl: String, credentials: LoginCredentials)
+    suspend fun updateCredentials(rawUrl: String, credentials: LoginCredentials)
 
     suspend fun getAllCredentials(): Flow<List<LoginCredentials>>
 
     suspend fun deleteCredentials(id: Int)
     suspend fun updateCredentials(credentials: LoginCredentials)
+
+    suspend fun containsCredentials(rawUrl: String, username: String, password: String): ContainsCredentialsResult
+
+    sealed interface ContainsCredentialsResult {
```

Comment:
NIT: Maybe this is better as an enum since all classes here are objects.

Reply:
That's a good suggestion, but I'm tempted to keep it as `sealed` instead, though I don't have strong feelings either way.

1. If we do add further data to any of the result types, it's easier to do this with `sealed` approach
2. We tend to use `sealed` for results elsewhere too (I think more so than `enums`)

I'll leave it as is if there's no strong pushback